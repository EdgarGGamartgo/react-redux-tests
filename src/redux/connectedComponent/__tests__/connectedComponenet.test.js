import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import ConnectedComponent from './../connectedComponent'
import { buyCake } from './../../cake/cakeActions'

const mockStore = configureStore([])

describe('My Connected React-Redux Component', () => {

    let store
    let component

    beforeEach(() => {
        store = mockStore({
            cake: {
                numOfCakes: 1
            }
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <ConnectedComponent/>
            </Provider>
        )
    })

    

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findByType('button').props.onClick()
        })

        expect(store.dispatch).toHaveBeenCalledTimes(1)
        expect(store.dispatch).toHaveBeenCalledWith(
             buyCake()
         )
    })
})