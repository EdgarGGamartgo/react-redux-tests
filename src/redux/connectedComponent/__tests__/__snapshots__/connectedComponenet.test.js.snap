// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Array [
  <h2>
    Number of Cakes - 
    1
  </h2>,
  <button
    onClick={[Function]}
  >
    Buy Cakes!
  </button>,
]
`;
